namespaces/live.cloud-platform.service.justice.gov.uk/intranet-dev/resources/dms.tf

```diff
@@ -0,0 +1,106 @@
+  ssl_mode                    = data.kubernetes_secret.dms_secret.data.src_ssl
+
+  tags = {
+    Name        = "${var.team_name} Source Endpoint"
+    Description = "Managed by Terraform"
+    Application = var.application
+    Owner       = var.team_name
+    Env         = var.environment
+  }
+}
+
+resource "aws_dms_endpoint" "target" {
+  endpoint_id                 = "${var.team_name}-dst-${random_id.id.hex}"
+  endpoint_type               = "target"
+  engine_name                 = data.kubernetes_secret.dms_secret.data.dst_engine
+  extra_connection_attributes = ""
+  server_name                 = data.kubernetes_secret.dms_secret.data.dst_addr
+  database_name               = data.kubernetes_secret.dms_secret.data.dst_database
+  username                    = data.kubernetes_secret.dms_secret.data.dst_user
+  password                    = data.kubernetes_secret.dms_secret.data.dst_pass
+  port                        = data.kubernetes_secret.dms_secret.data.dst_port
+  ssl_mode                    = data.kubernetes_secret.dms_secret.data.dst_ssl
+
+  tags = {
+    Name        = "${var.team_name} Destination Endpoint"
+    Description = "Managed by Terraform"
+    Application = var.application
+    Owner       = var.team_name
+    Env         = var.environment
+  }
+}
+
+resource "aws_dms_replication_task" "replication_task" {
+  # two values make sense here: full-load or full-load-and-cdc; see https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Task.CDC.html for details
+  # most notably, Azure SQL does not support CDC
+  migration_type           = "full-load-and-cdc"
+  replication_instance_arn = module.test_dms.replication_instance_arn
+  replication_task_id      = "${var.team_name}-repl-${random_id.id.hex}"
+
+  source_endpoint_arn = aws_dms_endpoint.source.endpoint_arn
+  target_endpoint_arn = aws_dms_endpoint.target.endpoint_arn
+
+  table_mappings            = "{\"rules\":[{\"rule-type\":\"selection\",\"rule-id\":\"1\",\"rule-name\":\"1\",\"object-locator\":{\"schema-name\":\"%\",\"table-name\":\"%\"},\"rule-action\":\"include\"}]}"
+  replication_task_settings = ""
+
+  # bug https://github.com/hashicorp/terraform-provider-aws/issues/1513
+  lifecycle { ignore_changes = ["replication_task_settings"] }
+
+  tags = {
+    Name        = "${var.team_name} Replication Task"
+    Description = "Managed by Terraform"
+    Application = var.application
+    Owner       = var.team_name
+    Env         = var.environment
+  }
+}
+
+resource "kubernetes_secret" "dms_instance" {
+  metadata {
+    name      = "dms-instance"
+    namespace = var.namespace
+  }
+
+  data = {
+    replication_instance_arn = module.test_dms.replication_instance_arn
+    source                   = aws_dms_endpoint.source.endpoint_arn
+    destination              = aws_dms_endpoint.target.endpoint_arn
+    task                     = aws_dms_replication_task.replication_task.replication_task_arn
+  }
+}
```

Comment:
File has no new line at the end

Reply:
Some background into the consideration:
- https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log
- https://medium.com/@alexey.inkin/how-to-force-newline-at-end-of-files-and-why-you-should-do-it-fdf76d1d090e